cmake/Modules/FindComputeCpp.cmake

```diff
@@ -494,12 +494,39 @@ function(add_sycl_to_target)
   # This has the same effect as:
   # target_link_libraries(${SNN_ADD_SYCL_TARGET}
   #   PUBLIC -Wl,--allow-shlib-undefined ComputeCpp::ComputeCpp
-  # )
-  set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
-    LINK_LIBRARIES -Wl,--allow-shlib-undefined ComputeCpp::ComputeCpp
-  )
-  set_property(TARGET ${SNN_ADD_SYCL_TARGET} APPEND PROPERTY
-    INTERFACE_LINK_LIBRARIES -Wl,--allow-shlib-undefined ComputeCpp::ComputeCpp
-  )
+  # )  
+  if (MSVC)
+    # Allows compute++ to parse the Windows headers
+    list(APPEND COMPUTECPP_USER_FLAGS "-D_ALLOW_COMPILER_AND_STL_VERSION_MISMATCH=1")
```

Comment:
I would prefer this to be moved out of the `build_ir` function, maybe after [L177](https://github.com/codeplaysoftware/SYCL-DNN/blob/50c3a38e79646b1dae25337d1642e42c826c0d15/cmake/Modules/FindComputeCpp.cmake#L177), and it should probably be `COMPUTECPP_DEVICE_COMPILER_FLAGS` rather than `COMPUTECPP_USER_FLAGS`.